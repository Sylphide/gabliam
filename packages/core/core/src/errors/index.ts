export * from './value-validation-error';
export * from './parser-not-supported-error';
export * from './loader-config-pgk-not-installed-error';
