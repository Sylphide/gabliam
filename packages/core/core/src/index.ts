import 'reflect-metadata';
export {
  decorate,
  inject,
  injectable,
  interfaces as inversifyInterfaces,
  multiInject,
  named,
  optional,
  postConstruct,
  tagged,
  targetName,
  unmanaged,
} from 'inversify';
export {
  APP_CONFIG,
  CORE_CONFIG,
  INJECT_CONTAINER_KEY,
  VALUE_EXTRACTOR,
} from './constants';
export * from './container';
export * from './decorator';
export * from './errors';
export * from './gabliam';
export * from './interfaces';
export * from './joi';
export { configResolver, FileLoader, Resolver } from './loaders';
export * from './metadata';
export * from './promise-utils';
export * from './reflection';
export * from './registry';
export * from './utils';
