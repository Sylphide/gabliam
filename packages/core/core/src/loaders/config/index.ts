export * from './config-resolver';
export * from './file-loader';
export * from './loader-config';
