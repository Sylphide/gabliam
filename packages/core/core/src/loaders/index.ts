export * from './loader-config';
export * from './loader-module';
export * from './config-resolver';
export { default as FileLoader } from './file-loader';
