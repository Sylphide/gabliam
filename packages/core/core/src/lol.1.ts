import 'reflect-metadata';
import { makePropDecorator } from './decorator/util';
import { Reflection } from './reflection/reflection';
import { interfaces } from 'inversify';

export interface BeanDecorator {
  (id: interfaces.ServiceIdentifier<any>): any;

  new (id: interfaces.ServiceIdentifier<any>): any;
}

// tslint:disable-next-line:no-empty-interface
export interface Bean {
  id: interfaces.ServiceIdentifier<any>;
}

export const Bean: BeanDecorator = makePropDecorator(
  'Bean',
  (id: interfaces.ServiceIdentifier<any>) => ({ id })
);

export interface LolDecorator {
  (): any;

  new (): any;
}

export const Lol: LolDecorator = makePropDecorator('Lol');

class Test /* extends TestLol */ {
  @Bean('lol2')
  @Bean('lol')
  @Lol()
  lol() {}
}

const reflection = new Reflection();

console.log('reflection test', reflection.parameters(Test, 'lol'));

console.log('reflexion props', reflection.propMetadata(Test));

console.log(
  'reflexion getPropsOfMetadata by name',
  reflection.getPropsOfMetadata(Test, 'Bean')
);

console.log(
  'reflexion getPropsOfMetadata by deco',
  reflection.getPropsOfMetadata(Test, Bean)
);
// console.log(require('util').inspect((t as any)[PARAMETERS][0][0]), {depth: null});
