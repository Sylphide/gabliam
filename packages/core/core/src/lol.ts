import 'reflect-metadata';
import {
  makeParamDecorator,
  TypeDecorator,
  makePropDecorator,
} from './decorator/util';
import { Reflection } from './reflection/reflection';

export interface ParamDecorator {
  (obj?: Param): TypeDecorator;

  new (obj?: Param): Param;
}

export interface Param {
  name?: string;
}

export const Context: ParamDecorator = makeParamDecorator(
  'Context',
  (p: Param = {}) => ({ ...p })
);

export interface LolDecorator {
  (): any;

  new (): any;
}

export const Lol: LolDecorator = makePropDecorator('Lol');

// class TestLol {
//   @Lol()
//   lolilol() {

//   }
// }

class Test /* extends TestLol */ {
  @Lol()
  lol() {}
  test(@Context({ name: 'lol' }) lol: string, @Context() context: string) {}

  test2(
    @Context() lol: string,
    @Context({ name: 'context' }) context: string
  ) {}
}

const reflection = new Reflection();

console.log('reflection test', reflection.parameters(Test, 'test'));

console.log('reflexion props', reflection.propMetadata(Test));
// console.log(require('util').inspect((t as any)[PARAMETERS][0][0]), {depth: null});
