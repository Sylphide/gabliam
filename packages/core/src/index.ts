import 'reflect-metadata';
export * from './interfaces';
export * from './decorators';
export * from './gabliam';
export {
  APP_CONFIG,
  CORE_CONFIG,
  VALUE_EXTRACTOR,
  INJECT_CONTAINER_KEY,
} from './constants';

export * from './registry';
export * from './errors';
export * from './utils';
export * from './joi';
export * from './container';
export { FileLoader, Resolver, configResolver } from './loaders';

export {
  injectable,
  tagged,
  named,
  inject,
  optional,
  unmanaged,
  multiInject,
  targetName,
  decorate,
  postConstruct,
  interfaces as inversifyInterfaces,
} from 'inversify';
