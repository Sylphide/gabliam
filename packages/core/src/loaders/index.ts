export * from './loader-config';
export * from './loader-module';
export { default as FileLoader } from './file-loader';
