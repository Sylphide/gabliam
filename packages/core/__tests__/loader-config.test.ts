// tslint:disable:one-line
// tslint:disable:no-unused-expression
import { LoaderConfig, FileLoader } from '../src/loaders';
// import * as mock from 'mock-fs';
import { MockFs } from './mock-fs';
import * as path from 'path';

let loader: LoaderConfig;
beforeEach(async () => {
  loader = new LoaderConfig();
});

test(`with config folder doesn't exist`, async () => {
  const config = await loader.load(__dirname, 'doesntexist/config');
  expect(config).toMatchSnapshot();
});

test(`with bad application.yml`, async () => {
  const config = await loader.load(
    path.resolve(__dirname, './fixtures/badyml')
  );
  expect(config).toMatchSnapshot();
});

test(`with application.yml empty`, async () => {
  const mock = new MockFs({
    'test/config': {
      'application.yml': ``
    }
  });
  const config = await loader.load(mock.resolve('test/config'));
  expect(config).toMatchSnapshot();
  mock.restore();
});

test(`with application.yml with nothing`, async () => {
  const mock = new MockFs({
    'test/config': {
      // tslint:disable-next-line:no-trailing-whitespace
      'application.yml': `

              `
    }
  });
  const config = await loader.load(mock.resolve('test/config'));
  expect(config).toMatchSnapshot();
  mock.restore();
});

test(`with application.yml`, async () => {
  const config = await loader.load(
    path.resolve(__dirname, './fixtures/config')
  );
  expect(config).toMatchSnapshot();
});

describe('with application.yml', async () => {
  let mock: MockFs;
  beforeAll(async () => {
    mock = new MockFs({
      'test/config': {
        'application.yml': `
              application:
                host: 127.0.0.1
                db: test
              testPath: path:./test.json
            `,
        'application-prod.yml': `
              application:
                host: prod.app.com
            `
      }
    });
  });

  test('load application.yml', async () => {
    const config = await loader.load(mock.resolve('test/config'));
    expect(config).toMatchSnapshot();
  });

  test('load with bad profile', async () => {
    const config = await loader.load(mock.resolve('test/config'), 'int');
    expect(config).toMatchSnapshot();
  });

  test('load with prod profile', async () => {
    const config = await loader.load(mock.resolve('test/config'), 'prod');
    expect(config).toMatchSnapshot();
    // expect(config).toMatchSnapshot(<any>{
    //   testPath: expect.any(String)
    // });
  });

  afterAll(() => mock.restore());
});

describe('with application.yml with bad handler', async () => {
  let mock: MockFs;
  beforeAll(async () => {
    mock = new MockFs({
      'test/config': {
        'application.yml': `
              application:
                host: 127.0.0.1
                db: test
              testPath: file:./test.json
            `,
        'application-prod.yml': `
              application:
                host: prod.app.com
            `
      }
    });
  });

  test('must throw error', async () => {
    const config = loader.load(mock.resolve('test/config'), 'int');
    await expect(config).rejects.toMatchSnapshot();
  });

  afterAll(() => mock.restore());
});

test(`with application.json`, async () => {
  const config = await loader.load([
    {
      loader: FileLoader,
      options: {
        folder: path.resolve(__dirname, './fixtures/config'),
        types: ['json']
      }
    }
  ]);
  expect(config).toMatchSnapshot();
});

test(`with json and yml`, async () => {
  const config = await loader.load([
    {
      loader: FileLoader,
      options: {
        folder: path.resolve(__dirname, './fixtures/config'),
        types: ['json', 'yml']
      }
    }
  ]);
  expect(config).toMatchSnapshot();
});

test(`with yml and json`, async () => {
  const config = await loader.load([
    {
      loader: FileLoader,
      options: {
        folder: path.resolve(__dirname, './fixtures/config'),
        types: ['yml', 'json']
      }
    }
  ]);
  expect(config).toMatchSnapshot();
});

test(`with bad application.json`, async () => {
  const config = await loader.load([
    {
      loader: FileLoader,
      options: {
        folder: path.resolve(__dirname, './fixtures/badjson'),
        types: ['json']
      }
    }
  ]);
  expect(config).toMatchSnapshot();
});

test(`with application.json empty`, async () => {
  const mock = new MockFs({
    'test/config': {
      'application.json': ``
    }
  });
  const config = await loader.load([
    {
      loader: FileLoader,
      options: {
        folder: mock.resolve('test/config'),
        types: ['json']
      }
    }
  ]);
  expect(config).toMatchSnapshot();
  mock.restore();
});

test(`with application.json with nothing`, async () => {
  const mock = new MockFs({
    'test/config': {
      // tslint:disable-next-line:no-trailing-whitespace
      'application.json': `

              `
    }
  });
  const config = await loader.load([
    {
      loader: FileLoader,
      options: {
        folder: mock.resolve('test/config'),
        types: ['json']
      }
    }
  ]);
  expect(config).toMatchSnapshot();
  mock.restore();
});

test(`with bad parser`, async () => {
  const config = loader.load([
    {
      loader: FileLoader,
      options: {
        folder: path.resolve(__dirname, './fixtures/badparser'),
        types: ['badparser']
      }
    }
  ]);
  await expect(config).rejects.toMatchSnapshot();
});

test(`with bad loader`, async () => {
  const config = loader.load([
    {
      loader: 'BadLoader'
    }
  ]);
  await expect(config).rejects.toMatchSnapshot();
});
