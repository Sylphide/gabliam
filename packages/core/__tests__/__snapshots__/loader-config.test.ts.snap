// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`with application.json 1`] = `
Object {
  "application": Object {
    "host": "prod.app.json",
    "json": true,
    "mongoose": Object {
      "mongoose": Object {
        "url": "mongo://localhost",
      },
    },
  },
}
`;

exports[`with application.json empty 1`] = `Object {}`;

exports[`with application.json with nothing 1`] = `Object {}`;

exports[`with application.yml 1`] = `
Object {
  "application": Object {
    "host": "prod.app.yml",
    "mongoose": Object {
      "mongoose": Object {
        "url": "mongo://localhost",
      },
    },
    "yml": true,
  },
}
`;

exports[`with application.yml empty 1`] = `Object {}`;

exports[`with application.yml load application.yml 1`] = `
Object {
  "application": Object {
    "db": "test",
    "host": "127.0.0.1",
  },
  "testPath": "<TEMP_DIR>/mock-fs-test/test/config/test.json",
}
`;

exports[`with application.yml load with bad profile 1`] = `
Object {
  "application": Object {
    "db": "test",
    "host": "127.0.0.1",
  },
  "testPath": "<TEMP_DIR>/mock-fs-test/test/config/test.json",
}
`;

exports[`with application.yml load with prod profile 1`] = `
Object {
  "application": Object {
    "db": "test",
    "host": "prod.app.com",
  },
  "testPath": "<TEMP_DIR>/mock-fs-test/test/config/test.json",
}
`;

exports[`with application.yml with bad handler must throw error 1`] = `[Error: ENOENT: no such file or directory, open '<TEMP_DIR>/mock-fs-test/test/config/test.json']`;

exports[`with application.yml with nothing 1`] = `Object {}`;

exports[`with bad application.json 1`] = `Object {}`;

exports[`with bad application.yml 1`] = `Object {}`;

exports[`with bad loader 1`] = `[LoaderConfigPgkNotInstalledError: BadLoader package has not been found installed. Try to install it: npm install BadLoader --save or yarn add BadLoader]`;

exports[`with bad parser 1`] = `[ParserNotSupportedError: The parser "badparser" is not supported by Gabliam]`;

exports[`with config folder doesn't exist 1`] = `Object {}`;

exports[`with json and yml 1`] = `
Object {
  "application": Object {
    "host": "prod.app.yml",
    "json": true,
    "mongoose": Object {
      "mongoose": Object {
        "url": "mongo://localhost",
      },
    },
    "yml": true,
  },
}
`;

exports[`with yml and json 1`] = `
Object {
  "application": Object {
    "host": "prod.app.json",
    "json": true,
    "mongoose": Object {
      "mongoose": Object {
        "url": "mongo://localhost",
      },
    },
    "yml": true,
  },
}
`;
